### ISSUE TITLE
Errors from list-table get ignored

### ISSUE THREAD

**Issue by rickatnight11** [sphinx-doc/sphinx#4132]
### Problem
Errors when parsing `list-table` directives don't seem to be treated like other errors (don't abort the build or exit non-zero).

#### Procedure to reproduce the problem

Test RST:

```
Test!
=====

.. list-table::
   :header-rows: 1

   * - Col1
     - Col2
     - Column3
   * - Some data 
     - Some more data
     - Even more data!
   * - Hey, here's another row 
     - With some
     - more data!
```

Builds cleanly:

```
[rwhite@fedorabot docs]$ make html
sphinx-build -b html -t ptinternal -d _build/doctrees   . _build/html
Running Sphinx v1.5.2
loading pickled environment... done
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 1 source files that are out of date
updating environment: 0 added, 0 changed, 0 removed
looking for now-outdated files... none found
preparing documents... done
writing output... [100%] index                                                                                                                                                                                                                                                                                                
generating indices... genindex
writing additional pages... search
copying static files... WARNING: favicon file '_static/favicon.ico' does not exist
done
copying extra files... done
dumping search index in English (code: en) ... done
dumping object inventory... done
build succeeded, 1 warning.

Build finished. The HTML pages are in _build/html.
[rwhite@fedorabot docs]$ echo $?
0
```

Then break with an indentation change:

```
Test!
=====

.. list-table::
   :header-rows: 1

   * - Col1
     - Col2
     - Column3
    * - Some data 
     - Some more data
     - Even more data!
   * - Hey, here's another row 
     - With some
     - more data!
```

Build shows the error, but does not abort or exit non-zero, and follow-up build pretends to be clean:

```
[rwhite@fedorabot docs]$ make html
sphinx-build -b html -t ptinternal -d _build/doctrees   . _build/html
Running Sphinx v1.5.2
loading pickled environment... done
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 1 source files that are out of date
updating environment: 0 added, 1 changed, 0 removed
reading sources... [100%] index                                                                                                                                                                                                                                                                                               
/home/rwhite/repos/sphinx-test/docs/index.rst:10: WARNING: Bullet list ends without a blank line; unexpected unindent.
/home/rwhite/repos/sphinx-test/docs/index.rst:11: WARNING: Bullet list ends without a blank line; unexpected unindent.
/home/rwhite/repos/sphinx-test/docs/index.rst:13: WARNING: Block quote ends without a blank line; unexpected unindent.
/home/rwhite/repos/sphinx-test/docs/index.rst:4: ERROR: Error parsing content block for the "list-table" directive: exactly one bullet list expected.

.. list-table::
   :header-rows: 1

   * - Col1
     - Col2
     - Column3
    * - Some data
     - Some more data
     - Even more data!
   * - Hey, here's another row
     - With some
     - more data!
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] index                                                                                                                                                                                                                                                                                                
generating indices... genindex
writing additional pages... search
copying static files... WARNING: favicon file '_static/favicon.ico' does not exist
done
copying extra files... done
dumping search index in English (code: en) ... done
dumping object inventory... done
build succeeded, 5 warnings.

Build finished. The HTML pages are in _build/html.
[rwhite@fedorabot docs]$ echo $?
0
[rwhite@fedorabot docs]$ make html
sphinx-build -b html -t ptinternal -d _build/doctrees   . _build/html
Running Sphinx v1.5.2
loading pickled environment... done
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 0 source files that are out of date
updating environment: 0 added, 0 changed, 0 removed
looking for now-outdated files... none found
no targets are out of date.
build succeeded.

Build finished. The HTML pages are in _build/html.
```

#### Expected results
Treat these errors like any other errors by aborting the build and exiting non-zero.

### Environment info
- OS: Fedora 26
- Python version: 2.7.13
- Sphinx version: 1.5.2

**Comment by tk0miya** [comment]
Thank you for reporting.
In rules of Sphinx, the error message is wrong. It is treated as a warning internally.
I just fixed message at cfadf17.

Please use `-W` option to handle it as an error.

**Comment by rickatnight11** [comment]
Wonderful, thanks!  We had other warning messages we did want to ignore, so it's good to get this fixed!
